### PR TITLE
Clean DiskANN log and fix wrong info in output

### PR DESCRIPTION
**Pull request by matchyc**
Signed-off-by: matchyc <dawnlight.yc@protonmail.com>

related: #253 

1. Use `easylogging++` to replace `diskann::cout` in most situation. Set different log levels for outputs.
 - Use `build.sh -t Release` to hide annoying debug info.
2. Fix wrong output when building (alpha factor).
3. Remove redundant files when using `merged vamana` index.

Disable original diskann executable files for quicker building (let me know if we need them).

### CODE REVIEW

File: thirdparty/DiskANN/CMakeLists.txt [hunk 1]
```diff
@@ -229,10 +229,15 @@ else()
...
+	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -Ofast -DELPP_DISABLE_DEBUG_LOGS -DNDEBUG -march=native -mtune=native -ftree-vectorize")
 	add_compile_options(-march=native -Wall -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free -fopenmp -fopenmp-simd -funroll-loops -Wfatal-errors -DUSE_AVX2)
 endif()
 
+include_directories( ${KNOWHERE_SOURCE_DIR} )
+include_directories( ${KNOWHERE_SOURCE_DIR}/thirdparty/ )
+add_definitions( -DAUTO_INITIALIZE_EASYLOGGINGPP )
+
 add_subdirectory(src)
-add_subdirectory(tests)
```
Comment (liliu-z) [inline thirdparty/DiskANN/CMakeLists.txt:237]: I think we can have another PR for this, and also delete the whole test folder is fine to me.

File: thirdparty/DiskANN/include/utils.h [hunk 2]
```diff
@@ -596,25 +590,24 @@ namespace diskann {
              << " while expected size is  " << expected_actual_file_size
              << " npts = " << npts << " dim = " << dim
              << " size of <T>= " << sizeof(T) << std::endl;
-      diskann::cout << stream.str() << std::endl;
+      LOG(ERROR) << stream.str();
```
Comment (liliu-z) [inline thirdparty/DiskANN/include/utils.h:593]: Why we need stream here?

File: thirdparty/DiskANN/src/index.cpp [hunk 3]
```diff
@@ -1714,11 +1713,11 @@ namespace diskann {
                              Parameters &             parameters,
                              const std::vector<TagT> &tags) {
     if (!file_exists(filename)) {
-      diskann::cerr << "Data file " << filename
-                    << " does not exist!!! Exiting...." << std::endl;
+      // diskann::cerr << "Data file " << filename
+                    // << " does not exist!!! Exiting...." << std::endl;
```
Comment (liliu-z) [inline thirdparty/DiskANN/src/index.cpp:1717]: Why comment them, we can delete it ?

File: thirdparty/DiskANN/src/index.cpp [hunk 4]
```diff
@@ -1714,11 +1713,11 @@ namespace diskann {
...
       std::stringstream stream;
       stream << "Data file " << filename << " does not exist." << std::endl;
-      diskann::cerr << stream.str() << std::endl;
```
Comment (liliu-z) [inline thirdparty/DiskANN/src/index.cpp:1721]: Same, do we need stream here?

File: thirdparty/DiskANN/CMakeLists.txt (hunk 1 above)
Comment (matchyc) [reply thirdparty/DiskANN/CMakeLists.txt:237]: Will exclude this.

File: thirdparty/DiskANN/src/index.cpp [hunk 5]
```diff
@@ -1609,17 +1610,16 @@ namespace diskann {
       MallocExtension::instance()->ReleaseFreeMemory();
 #endif
       if (_nd > 0) {
-        diskann::cout << "Completed Pass " << rnd_no << " of data using L=" << L
-                      << " and alpha=" << parameters.Get<float>("alpha")
-                      << ". Stats: ";
-        diskann::cout << "search+prune_time=" << total_sync_time
+        LOG(INFO) << "Completed Pass " << rnd_no << " of data using L=" << L
```
[the remark this answers is not in the document]
Comment (matchyc) [reply thirdparty/DiskANN/src/index.cpp:1613]: Only 2 lines for each building process. Not the massive calculation step.

[the remark this answers is not in the document]
Comment (liliu-z) [reply thirdparty/DiskANN/src/index.cpp:1613]: sure, we can find a strategy later

File: thirdparty/DiskANN/src/index.cpp [hunk 6]
```diff
@@ -1587,13 +1587,14 @@ namespace diskann {
...
-                        << "; inter_time: " << inter_time << "s" << std::endl;
-
+          std::stringstream stream;
+          stream.precision(4);
+          stream << "Completed  (round: " << rnd_no
+                 << ", sync: " << sync_num << "/" << num_syncs
+                 << " with L " << L << ")"
+                 << " sync_time: " << sync_time << "s"
+                 << "; inter_time: " << inter_time << "s" << std::endl;
+          LOG(DEBUG) << stream.str();
```
[the remark this answers is not in the document]
Comment (matchyc) [reply thirdparty/DiskANN/src/index.cpp:1597]: From documentation of easylogging, no method to set precision for now. This stream only use to prepare string for `LOG`.

File: thirdparty/DiskANN/src/index.cpp (hunk 3 above)
Comment (matchyc) [reply thirdparty/DiskANN/src/index.cpp:1717]: Deleted.

File: thirdparty/DiskANN/src/index.cpp (hunk 4 above)
Comment (matchyc) [reply thirdparty/DiskANN/src/index.cpp:1721]: It's from the original code of diskann. Just change the output approach from `cout` to `LOG`

File: thirdparty/DiskANN/include/utils.h (hunk 2 above)
Comment (matchyc) [reply thirdparty/DiskANN/include/utils.h:593]: From original diskann, just for concatenating the strings that we need.

File: thirdparty/DiskANN/include/utils.h [hunk 7]
```diff
@@ -596,25 +590,24 @@ namespace diskann {
...
       throw diskann::ANNException(stream.str(), -1, __FUNCSIG__, __FILE__,
                                   __LINE__);
     }
     rounded_dim = ROUND_UP(dim, 8);
-    diskann::cout << "Metadata: #pts = " << npts << ", #dims = " << dim
-                  << ", aligned_dim = " << rounded_dim << "... " << std::flush;
+    std::stringstream stream;
```
[the remark this answers is not in the document]
Comment (matchyc) [reply thirdparty/DiskANN/include/utils.h:598]: Unused, but will use it for concatenating strings later.

**Comment by liliu-z** [comment]
/approve

Thanks for this PR!

**Review by liliu-z — APPROVED** [review]

**Pull request merged**